[DependencyInjection] Document target parameter for #[AsAlias] attribute (#21826)
symfony/symfony-docs: service_container/alias_private.rst

            // ...
        }

You can also use the ``target`` parameter to create aliases for specific
:ref:`named autowiring <autowiring-multiple-implementations-same-type>` targets.
This is useful when you have multiple implementations of the same interface

> and want to inject a specific one based on the argument name::

Reviewer: ```suggestion
and need to inject a specific one based on context.

For example, you might have different mailer implementations for different types of emails::
```

    // src/Mail/PhpMailer.php
    namespace App\Mail;

    use Symfony\Component\DependencyInjection\Attribute\AsAlias;

>     #[AsAlias(MailerInterface::class, target: 'mainMailer')]

Reviewer: ```suggestion
    #[AsAlias(MailerInterface::class, target: 'notificationMailer')]
```

>     class PhpMailer implements MailerInterface

Reviewer: ```suggestion
    class SmtpMailer implements MailerInterface
```

    {
        // ...
    }

> 

Reviewer: ```suggestion
::
```

    // src/Mail/SendGridMailer.php
    namespace App\Mail;

    use Symfony\Component\DependencyInjection\Attribute\AsAlias;

    #[AsAlias(MailerInterface::class, target: 'transactionalMailer')]

>     class SendGridMailer implements MailerInterface

Reviewer: ```suggestion
    class TransactionalMailer implements MailerInterface
```

    {
        // ...
    }

Now you can inject the specific implementation using the target name with
the :ref:`#[Target] attribute <autowiring-multiple-implementations-same-type>`::

    use Symfony\Component\DependencyInjection\Attribute\Target;

    class NewsletterService
    {
        public function __construct(
            #[Target('mainMailer')]
            private MailerInterface $mailer,
        ) {
        }
    }

>     class OrderService

Reviewer: ```suggestion
    class OrderConfirmationService
```

    {
        public function __construct(
            #[Target('transactionalMailer')]
            private MailerInterface $mailer,
        ) {
        }
    }

> 

Reviewer: You can then inject the specific implementation by matching the argument name
to the target::

    class UserNotificationService
    {
        public function __construct(
            private MailerInterface $notificationMailer,
        ) {
        }
    }

::

    class OrderConfirmationService
    {
        public function __construct(
            private MailerInterface $transactionalMailer,
        ) {
        }
    }

Alternatively, you can use the :ref:`#[Target] attribute <autowiring-multiple-implementations-same-type>`
to be more explicit::

    use Symfony\Component\DependencyInjection\Attribute\Target;

    class UserNotificationService
    {
        public function __construct(
            #[Target('notificationMailer')]
            private MailerInterface $mailer,
        ) {
        }
    }

::

    class OrderConfirmationService
    {
        public function __construct(
            #[Target('transactionalMailer')]
            private MailerInterface $mailer,
        ) {
        }
    }

            // ...
        }

You can also use the ``target`` parameter to create aliases for specific
:ref:`named autowiring <autowiring-multiple-implementations-same-type>` targets.
This is useful when you have multiple implementations of the same interface
and need to inject a specific one based on context.

For example, you might have different mailer implementations for different types of emails::

    // src/Mail/PhpMailer.php
    namespace App\Mail;

    use Symfony\Component\DependencyInjection\Attribute\AsAlias;

    #[AsAlias(MailerInterface::class, target: 'notificationMailer')]
    class SmtpMailer implements MailerInterface
    {
        // ...
    }

> ::

Reviewer 2: I guess its possible to define those two classes in same codeblock as in [this page](https://symfony.com/doc/current/service_container.html#limiting-services-to-a-specific-symfony-environment) while removing file name and namespace
